src/core/evaluator.js

```diff
@@ -402,6 +403,25 @@ class PartialEvaluator {
     return data;
   }
 
+  loadFontInDocument(name) {
+    const cachedData = this.standardFontDataCache.get(name);
+    if (cachedData) {
+      return cachedData;
+    }
+    const subst = getFontSubstitution(
+      this.idFactory.getDocId(),
+      this.options.standardFontDataUrl,
+      name
+    );
+    if (!subst) {
+      return null;
+    }
+    const loadedName = subst.args[0];
+    this.standardFontDataCache.set(name, loadedName);
+    this.handler.send("LoadFontInDocument", subst.args);
+    return subst.cssString;
```

Comment:
Sorry, but I *really* don't like all of the special-handling introduced here and *especially* on the main-thread for this feature!

Why can't we just include/export this data as a regular property on the font, and then handle that on the main-thread instead?
The following `diff` obviously won't work as-is, but to try and illustrate my point of re-using our existing coding-patterns:
```diff
diff --git a/src/display/font_loader.js b/src/display/font_loader.js
index 307c68838..5a5ae0695 100644
--- a/src/display/font_loader.js
+++ b/src/display/font_loader.js
@@ -73,11 +73,15 @@ class FontLoader {

   async bind(font) {
     // Add the font to the DOM only once; skip if the font is already loaded.
-    if (font.attached || font.missingFile) {
+    if (font.attached || font.missingFile || !font.cssFontFamily) {
       return;
     }
     font.attached = true;

+    if (font.cssFontFamily) {
+      return this.loadFont(font.cssFontFamily); // <-- This calls your new method.
+    }
+
     if (this.isFontLoadingAPISupported) {
       const nativeFontFace = font.createNativeFontFace();
       if (nativeFontFace) {
```

The advantage of doing something along these lines is that we don't need to make any changes in the `src/display/api.js` file.

Reply:
My idea was to send the load message as soon as possible on the main thread.
But when looking more in details it seems that there is not that much time.